core/src/test/java/org/testcontainers/containers/GenericContainerTest.java

```diff
@@ -21,6 +24,9 @@
 
     @Test
     public void shouldReportOOMAfterWait() {
+        Info info = DockerClientFactory.instance().client().infoCmd().exec();
+        // Poor man's rootless Docker detection :D
```

Comment:
😂 for our tests, could we do this using the host socket URI?

Reply:
I didn't want to do string matching, especially given that the feature is experimental and may change the socket location.

Apparently, Docker returns `rootless` security capability from the info endpoint, just docker-java does not expose it (yet). I will add it to docker-java, so that we can remove this workaround, okay? :)

Reply:
Cool, I was wondering if that would be the case. Let's not hold up this PR for the change in docker-java, though - it's OK as-is.